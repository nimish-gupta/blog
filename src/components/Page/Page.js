import React, { useRef, useEffect } from 'react';
import * as Styled from './Page.style';

type Props = {
	title?: string,
	children: React.Node,
};

const Page = ({ title, children }: Props) => {
	const pageRef = useRef();

	useEffect(() => {
		pageRef.current.scrollIntoView();
	});

	return (
		<Styled.Page ref={pageRef}>
			<Styled.Inner>
				{title && <Styled.Title>{title}</Styled.Title>}
				<Styled.Body>{children}</Styled.Body>
			</Styled.Inner>
		</Styled.Page>
	);
};

export default Page;
