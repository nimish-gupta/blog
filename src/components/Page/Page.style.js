/* eslint-disable prefer-template */
import styled from 'styled-components';
import { math } from 'polished';

import {
	breakpointSm,
	breakpointMd,
	lineHeight,
	marginTop,
	marginBottom,
	margin,
} from '../../assets/styled-components/mixins';

export const Page = styled.div`
	${marginBottom(2)};
	${breakpointSm`
    float:right;
    width: 59%;
  `};
	${breakpointMd`
    float:right;
    width: 65.59%;
  `};
`;

export const Inner = styled.div`
	padding: 25px 20px;
	${breakpointSm`
    padding: 30px 20px;
  `};
	${breakpointMd`
    padding: 40px 35px;
  `};
`;

export const Title = styled.h1`
	font-size: ${(props) => math(props.theme.typographic.baseFontSize + '2.5')};
	font-weight: 600;
	${lineHeight(2)};
	${marginTop(0)};
	${marginBottom(1.45)};
`;

export const Body = styled.div`
	font-size: ${(props) => props.theme.typographic.baseFontSize};
	${lineHeight(1)};
	${margin(0, 0, 1)};
`;
