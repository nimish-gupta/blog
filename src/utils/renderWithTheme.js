import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import getTheme from '../assets/styled-components/_variables';

export default function renderWithTheme(component) {
	return renderer.create(
		<ThemeProvider theme={getTheme('light')}>{component}</ThemeProvider>
	);
}
