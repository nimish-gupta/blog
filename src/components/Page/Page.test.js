// @flow strict
import React from 'react';

import Page from './Page';
import renderWithTheme from '../../utils/renderWithTheme';

describe('Page', () => {
	const props = {
		children: 'test',
		title: 'test',
	};

	it('renders correctly', () => {
		const tree = renderWithTheme(<Page {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
