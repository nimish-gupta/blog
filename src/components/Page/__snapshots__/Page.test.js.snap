// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
<div
  className="sc-AxjAm lcHhHg"
>
  <div
    className="sc-AxirZ fpEbtS"
  >
    <h1
      className="sc-AxiKw gCCtAg"
    >
      test
    </h1>
    <div
      className="sc-AxhCb jKCdpE"
    >
      test
    </div>
  </div>
</div>
`;
